radicle/src/storage.rs

```diff
@@ -174,13 +176,17 @@ pub struct Remote<V> {
     pub id: PublicKey,
     /// Git references published under this remote, and their hashes.
     pub refs: SignedRefs<V>,
+    // TODO(finto): Why does this live here? They could be a delegate of one project and not of another?
     /// Whether this remote is of a project delegate.
     pub delegate: bool,
     /// Whether the remote is verified or not, ie. whether its signed refs were checked.
     verified: PhantomData<V>,
 }
 
 impl<V> Remote<V> {
+    // TODO(finto): This function seems out of place in the API for a couple of reasons:
+    // * The SignedRefs aren't guaranteed to be by the `id`
+    // * I could write `Remote::<Verified>::new(id, refs) and because of the above, it's a LIE
```

Comment:
Also, this looks like a leaky function, but a second pair of eyes might help confirm that :)